### PR TITLE
Migrated operators tests to `pytest`

### PR DESCRIPTION
**Pull request by Abhishek-kumar-ISM**
Migrated some tests/operators to pytest

related: https://github.com/apache/airflow/issues/29305

### CODE REVIEW

**Comment by Abhishek-kumar-samsung** [comment]
@Taragolis migrated operators tests to pytest.

Can you please review.

**Review by Taragolis — APPROVED** [review]

**Comment by Abhishek-kumar-samsung** [comment]
@Taragolis when an approved PR is finally merged by those who have write access?
when all checks pass?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
In general yes - there was some problem in this case with available built images (something to solve separately). I rerun it and sometime after tests are all green we merge it (if you notice we did not - ping us).

**Comment by potiuk** [comment]
In general yes - there was some problem in this case with available built images (something to solve separately). I rerun it and sometime after tests are all green we merge it (if you notice we did not - ping us).

**Comment by Abhishek-kumar-samsung** [comment]
@potiuk thankyou, i had one more PR related with the same issue.
https://github.com/apache/airflow/pull/29373
which was also approved, can you re run that also?

**Comment by eladkal** [comment]
> i had one more PR related with the same issue. #29373 which was also approved, can you re run that also?

rebase triggers new test run (I did it for the PR)

**Comment by Abhishek-kumar-samsung** [comment]
Okay thankyou.

**Comment by Abhishek-kumar-samsung** [comment]
@Taragolis migrated all tests/operators tests to pytest. Please check

**Comment by Abhishek-kumar-samsung** [comment]
@Taragolis @potiuk @eladkal please review and merge the PR.

File: tests/operators/test_datetime.py
Comment (Taragolis) [inline tests/operators/test_datetime.py]: There is two same parameters: `(datetime.time(10, 0, 0), datetime.time(11, 0, 0))`
That a good question what we actually expected when we pass two `datetime.time` arguments and `target_lower >  target_upper` it could refers to the fact that BranchOperator work wrong with this case.

Currently this test pass due to the fact that subtests do not reset DagRuns and other stuff and use only first one, it is not a case for `@pytest.mark.parametrize` and I guess initially this test failed

File: tests/operators/test_datetime.py
Comment (Taragolis) [inline tests/operators/test_datetime.py]:
```suggestion
        targets,
```

File: tests/operators/test_datetime.py
Comment (Taragolis) [inline tests/operators/test_datetime.py]:
```suggestion
        targets,
```

File: tests/operators/test_datetime.py
Comment (Taragolis) [inline tests/operators/test_datetime.py]:
```suggestion
        targets,
```

**Review by Taragolis — CHANGES_REQUESTED** [review]

File: tests/operators/test_datetime.py
Comment (Abhishek-kumar-samsung) [reply tests/operators/test_datetime.py:50]: Yes when target_lower>target_upper, then some error was coming with the same issue that target_lower>target_upper.

I don't know why with subTests this error was not coming.

So i thought it was some error as lower time should be smaller.

Did i missed something?

Earlier one was correct? or i did wrong on making target_lower greater?

@Taragolis

Comment (Abhishek-kumar-ISM) [reply tests/operators/test_datetime.py]: @Taragolis 
so that was to check testcase when some wrong dates are passed?

I did wrong by changing target_lower?

Comment (Taragolis) [reply tests/operators/test_datetime.py]: > I did wrong by changing target_lower?

I would suggest just drop latest parameter, because it duplicated. I can't imagine how we could resolve `target_lower >  target_upper` in case of `datetime.time`.

This operator originally have a broken test scenario and that is why it pass in `main` branch now 🤣

**Comment by Abhishek-kumar-samsung** [comment]
@Taragolis done changes, as requested by you.

**Comment by Taragolis** [comment]
I will have a look later. 

Seem like that all operators tests migrated to `pytest` if so, could you also change a PR name?

**Comment by Abhishek-kumar-samsung** [comment]
> I will have a look later.
> 
> Seem like that all operators tests migrated to `pytest` if so, could you also change a PR name?

Done

**Review by Taragolis — APPROVED** [review]

**Pull request merged**
